File: Algospot/FESTIVAL/FESTIVAL.py
import sys
input = sys.stdin.readline

C = int(input())

for _ in range(C):
    N, L = map(int, input().split())
    dailyCost = list(map(int, input().split()))
    minRentCost = 987654321.0

    for i in range(N - L + 1):
        day = 1
        rentCost = 0
        for j in range(i, N):
            rentCost += dailyCost[j]

            if day >= L:
                if minRentCost > rentCost / day:
                    minRentCost = rentCost / day

            day += 1

    print("%.8f" % minRentCost)

File: Algospot/FESTIVAL/FESTIVAL.cpp
#include <iostream>

using namespace std;

int dailyCost[1001];

int main(void)
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    cout.setf(ios::fixed);
    cout.precision(8);

    int C = 0;
    cin >> C;

    int N = 0, L = 0;
    int day = 0;
    int rentCost = 0;
    double minRentCost = 0.0;
    while (C--)
    {
        cin >> N >> L;
        minRentCost = 987654321.0;

        for (int i = 0; i < N; i++)
            cin >> dailyCost[i];

        for (int i = 0; i <= N - L; i++)
        {
            day = 1;
            rentCost = 0;
            for (int j = i; j < N; j++)
            {
                rentCost += dailyCost[j];

                if (day >= L)
                {
                    if (minRentCost > rentCost / (double)day)
                        minRentCost = rentCost / (double)day;
                }

                day++;
            }
        }

        cout << minRentCost << "\n";
    }

    return 0;
}